pkg/sources/postman/postman.go

```diff
@@ -0,0 +1,678 @@
+	s.addKeyword(item.Name)
+
+	// override the base collection metadata with item-specific metadata
+	metadata.FolderID = item.ID
+	metadata.Type = FOLDER_TYPE
+	if metadata.FolderName != "" {
+		// keep track of the folder hierarchy
+		metadata.FolderName = metadata.FolderName + " > " + item.Name
+	} else {
+		metadata.FolderName = item.Name
+	}
+
+	if item.UID != "" {
+		metadata.FullID = item.UID
+		metadata.Link = LINK_BASE_URL + FOLDER_TYPE + "/" + metadata.FullID
+	}
+	// recurse through the folders
+	for _, subItem := range item.Items {
+		s.scanItem(ctx, chunksChan, collection, metadata, subItem)
+	}
+
+	// check if there are any requests in the folder
+	if item.Request.Method != "" {
+		metadata.RequestID = item.ID
+		metadata.RequestName = item.Name
+		metadata.Type = REQUEST_TYPE
+		if item.UID != "" {
+			// Route to API endpoint
+			metadata.FullID = item.UID
+			metadata.Link = LINK_BASE_URL + REQUEST_TYPE + "/" + item.UID
+		} else {
+			// Route to collection.json
+			metadata.FullID = item.ID
+		}
+		s.scanHTTPRequest(ctx, chunksChan, metadata, item.Request)
+	}
+
+	// check if there are any responses in the folder
+	for _, response := range item.Response {
+		s.scanHTTPResponse(ctx, chunksChan, metadata, response)
+	}
+
+	for _, event := range item.Events {
+		s.scanEvent(ctx, chunksChan, metadata, event)
+	}
+
+	// an auth all by its lonesome could be inherited to subfolders and requests
+	s.scanAuth(ctx, chunksChan, metadata, item.Auth, item.Request.URL)
+}
+
+func (s *Source) scanEvent(ctx context.Context, chunksChan chan *sources.Chunk, metadata Metadata, event Event) {
+	metadata.Type = metadata.Type + " > event"
+	data := strings.Join(event.Script.Exec, " ")
+
+	// Prep direct links. Ignore updating link if it's a local JSON file
+	if !metadata.fromLocal {
+		metadata.Link = LINK_BASE_URL + metadata.Type + "/" + metadata.FullID
+		if event.Listen == "prerequest" {
+			metadata.Link += "?tab=pre-request-scripts"
+		} else {
+			metadata.Link += "?tab=tests"
+		}
+	}
+
+	s.scanData(ctx, chunksChan, s.formatAndInjectKeywords(s.buildSubstitueSet(metadata, data)), metadata)
+}
+
+func (s *Source) scanAuth(ctx context.Context, chunksChan chan *sources.Chunk, m Metadata, auth Auth, u URL) {
+	var authData string
+	switch auth.Type {
+	case "apikey":
+		var apiKeyValue, apiKeyName string
+		for _, kv := range auth.Apikey {
+			switch kv.Key {
+			case "key":
+				apiKeyValue = fmt.Sprintf("%v", kv.Value)
+			case "value":
+				apiKeyName = fmt.Sprintf("%v", kv.Value)
+			}
+		}
+		authData += fmt.Sprintf("%s=%s\n", apiKeyName, apiKeyValue)
+	case "awsSigV4", "awsv4":
+		for _, kv := range auth.AWSv4 {
+			switch kv.Key {
+			case "accessKey":
+				authData += fmt.Sprintf("accessKey:%s ", kv.Value)
+			case "secretKey":
+				authData += fmt.Sprintf("secretKey:%s ", kv.Value)
+			case "region":
+				authData += fmt.Sprintf("region:%s ", kv.Value)
+			case "service":
+				authData += fmt.Sprintf("service:%s ", kv.Value)
+			}
+		}
```

Comment:
Why do we only take the last kv in the `"apikey"` case, but take all of them for the `"awsv4"` case? Why is that different? Same goes for the other cases but I wanted to highlight this example.

Reply:
```
			"auth": {
				"type": "basic",
				"basic": [
					{
						"key": "username",
						"value": "parentUsername",
						"type": "string"
					},
					{
						"key": "password",
						"value": "parentPassword",
						"type": "string"
					}
				]
			}
```

We have to iterate through KV elements and build up the data to be scanned.

Reply:
I guess I'm confused why the `"apikey"` case ignores everything but the last value in the array.